.github/workflows/check_python_dependencies.yml

```diff
@@ -0,0 +1,50 @@
+name: Check Python Dependencies
+
+on:
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  check_dependencies:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: ["3.12"]
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Clone repo-tools repo
+        uses: actions/checkout@v4
+        with:
+          repository: salman2013/repo-tools
+          path: repo_tools
+          ref: salman/add-script-for-python-dependencies
+                    
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      
+      - name: Install setuptool
+        run: pip install setuptools
+      
+      - name: Install requests module
+        run: pip install requests
+
+      - name: Install rich module
+        run: pip install rich
+
+      - name: Install requirements parser module
+        run: pip install requirements-parser  
```

Comment:
```suggestion                    
      - name: Set up Python
        uses: actions/setup-python@v5
        with:
          python-version: ${{ matrix.python-version }}

      - name: Install repo-tools
        run: pip install git+https://github.com/salman2013/repo-tools.git@salman/add-script-for-python-dependencies
```

Reply:
@feanil There is one thing in this, i need to install `requirements-parser` and  `setuptools` so there are two options should i add them here in CI or add them in repo-tool so that the script can find these packages. What do you suggest?